File: PN23.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        int iCnt = 0;
		
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        void DisplayF()      
        {
			 for(iCnt = 1; iCnt<= iNo; iCnt++)
             {
                 cout<<iCnt<<endl;        
             }
        }

        int CountFactor()      
        {
			int iFactCnt = 0;
			 for(iCnt = 1; iCnt<= (iNo/2); iCnt++)
             {
                 if(iNo % iCnt == 0)
				 {
					 iFactCnt++;
				 }					 
             }
			 return iFactCnt;
        }

};

int main()
{
    Number nobj;
	int iRet = 0;


    nobj.Accept();
    nobj.DisplayF();

    iRet = nobj.CountFactor();
	cout<<iRet<<endl;


    return 0;
}

File: PN23.java
// Template for probplems on Numbers

import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }
    
	public void DisEven()
	{
     int iDigit = 0;
	 int iCnt = 0;
	
	 if(iNo < 0)
	 {
		iNo = -iNo;
	 }
	 while(iNo > 0)
	 {
		iDigit = iNo % 10;
		if((iDigit %2 ) == 0)
		{
			System.out.println(+iDigit);
		
	    }
	    iNo = iNo/10;
	 }
	}
}

class PN23
{
    public static void main(String b[])
    {
            Number nobj = new Number();
           
			
            nobj.Accept();
            nobj.Display();

            nobj.DisEven();
			

    }
}